src/components/input-switch/input-switch.tsx

```diff
@@ -252,8 +252,10 @@ const InputSwitch = ({
           displayDateTimeLabels={displayDateTimeLabels}
         />;
       case 'integer2':
+      case 'int4':
```

Comment:
Instead of adding 2 new conditions, change the line above for the `ArrayField` definition:
`baseArrayType={columnModel?.column.type.baseType.name}`

To remove the `.name` so the component takes in an object instead. Make sure to refactor the array field component after making this change.